docs/design.md

```diff
@@ -0,0 +1,24 @@
+## High Level Deployment
```

Comment:
One objective section of the design can be helpful. For example, what is the intent of the operator and what is it we are trying to solve.